feature-libs/quote/root/guards/quote-cart.guard.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { inject, Injectable } from '@angular/core';
+import { CanActivate, UrlTree } from '@angular/router';
+
+import { Observable, combineLatest } from 'rxjs';
+import { QuoteCartService } from './quote-cart.service';
+import { map } from 'rxjs/operators';
+import { RouterState, RoutingService } from '@spartacus/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class QuoteCartGuard implements CanActivate {
+  protected routingService = inject(RoutingService);
+  protected quoteCartService = inject(QuoteCartService);
+
+  canActivate(): Observable<boolean | UrlTree> {
+    return combineLatest([
+      this.quoteCartService.isQuoteCartActive(),
+      this.quoteCartService.getQuoteId(),
+      this.quoteCartService.isCheckoutAllowed(),
+      this.routingService.getRouterState(),
+    ]).pipe(
+      map(([isQuoteCartActive, quoteId, isCheckoutAllowed, routerState]) => {
+        const isAllowedCheckoutNavigation = this.checkAllowedCheckoutNavigation(
+          routerState,
+          isCheckoutAllowed
+        );
+
+        if (isQuoteCartActive && !isAllowedCheckoutNavigation) {
+          this.routingService.go({
+            cxRoute: 'quoteDetails',
+            params: { quoteId: quoteId },
+          });
```

Comment:
[MINOR]
When redirecting inside Angular route guard, it's better to return the destination's UrlTree instead of true + performing an imperative navigation request.

Thanks to that, the internal router events show a `NavigationCanceled` (which is semantically a redirect event), instead of NavigationEnd + NavigationStart


```suggestion
          return this.router.parseUrl(
            this.semanticPathService.get({
              cxRoute: 'quoteDetails',
              params: { quoteId: quoteId },
            })
          });
```

note: `this.router.parseUrl()` converts a string to an UrlTree required by Angular route guard

Reply:
@ChristophHi